apps/openmw/mwmechanics/character.cpp

```diff
@@ -943,7 +943,7 @@ void CharacterController::handleTextKey(const std::string &groupname, const std:
         if(!sound.empty())
         {
             MWBase::SoundManager *sndMgr = MWBase::Environment::get().getSoundManager();
-            if(evt.compare(10, evt.size()-10, "left") == 0 || evt.compare(10, evt.size()-10, "right") == 0)
+            if(evt.compare(10, 4, "left") == 0 || evt.compare(10, 5, "right") == 0)
```

Comment:
Same here, it'll catch anything starting with left or right. Also I'm not sure if it's safe to specify an offset+length that can go past the end of the `evt` string.